### PR TITLE
Added support for phone, address and questions

### PR DESCRIPTION
**Pull request by aclarkd**
@todo feedback on the register button
@todo error on missing required params (projectId, clientId, etc)
@todo make configuation more obvious

### CODE REVIEW

File: src/LassoLead.php [hunk 1]
```diff
@@ -74,22 +107,85 @@ public function addEmail($email, $type = 'Home', $primary = true) {
...
+    ];
+  }
+
+  /*
+   * @param String $path folder structure
+   * @param String $question
+   * @param Mixed String|Array
+   */
+  public function addQuestion($path, $question, $answer) {
+    $answers = !is_array($answer) ? [['id' => '', 'text' => $answer]] : $answer;
```
Comment (trb) [inline src/LassoLead.php]: It's best not to use boolean inversions if not necessary (i.e. remove the '!')

File: signup.php [hunk 2]
```diff
@@ -4,34 +4,58 @@
...
+$projectId = '';
 
 /* This is a security token and should be kept private
  * this should not be a hidden field on your registration page.
  */
-$lassoUid  = '';
+$apiKey  = '';
+
+if(empty($clientId) || empty($projectId) || empty($apiKey)) {
+  throw new Exception('Required parameteres are not set, please
```
Comment (tutfoo) [inline signup.php]: typo: parameters

File: signup.php [hunk 3]
```diff
@@ -4,34 +4,58 @@
...
+                       check that your $clientId, $projectId and $apiKey are
+                       configured correctly');
+}
+
+/* Contructing and submitting a lead
```
Comment (tutfoo) [inline signup.php]: typo: Constructing

File: src/LassoLead.php [hunk 4]
```diff
@@ -74,22 +107,85 @@ public function addEmail($email, $type = 'Home', $primary = true) {
...
+   * @param String $address unit number
+   * @param String $city
+   * @param String $state state/province
+   * @param String $zip zip/postal code
+   * @param String $country
+   * @param String $type
+   * @param Boolean $primary
+   */
+  public function addAddress($address, $city, $state, $zip, $country, $type = 'Home', $primary = true) {
+    $this->addresses[] = [
```
Comment (tutfoo) [inline src/LassoLead.php:151]: The field "addresses" doesn't exist in this class

File: src/LassoLead.php [hunk 5]
```diff
@@ -74,22 +107,85 @@ public function addEmail($email, $type = 'Home', $primary = true) {
...
+      'address' => $address,
+      'city' => $city,
+      'country' => $country,
+      'province' => $state,
+      'postalCode' => $state,
```
Comment (tutfoo) [inline src/LassoLead.php]: $zip

File: src/LassoLead.php [hunk 6]
```diff
@@ -37,35 +37,68 @@ class LassoLead
...
+     }
+   */
+  private $phones = [];
+
+  /*
+   * Array the question objects
+   * {
+       email: '123 456 7890',
+       type: 'Mobile',
+       primary: true
```
Comment (tutfoo) [inline src/LassoLead.php]: comment block needs update

File: src/LassoLead.php [hunk 7]
```diff
@@ -37,35 +37,68 @@ class LassoLead
   private $emails = [];
 
   /*
-   * String the rating the registrant should be entered with
+   * Array the phone objects
+   * {
+       email: '123 456 7890',
+       type: 'Mobile',
+       primary: true
```
Comment (tutfoo) [inline src/LassoLead.php]: comment block needs update

**Comment by tutfoo** [comment]
comment blocks should start with a /*\* instead of a /\* to take advantage of PHPdoc

**Comment by tutfoo** [comment]
I will be taking over changes to this pull request

**Comment by tutfoo** [comment]
Please don't merge until squashed - if you're happy with these changes, let me know to squash first before you merge

File: signup.php [hunk 8]
```diff
@@ -3,35 +3,56 @@
 require('src/LassoLead.php');
 require('src/RegistrantSubmitter.php');
 
-/* These variables should only be attached to the request on the server side */
-$clientId = 19;
-$projectId = 6;
+/**
```
Comment (trb) [inline signup.php]: Use '/*' for comment blocks that aren't phpdoc

**Comment by tutfoo** [comment]
Fixed /*\* to /\* comments in signup.php and squashed.

File: src/LassoLead.php [hunk 9]
```diff
@@ -1,71 +1,122 @@
...
+   *   address: '123 Street',
+   *   city: 'Vancouver',
+   *   country: 'Canada',
+   *   province: 'BC',
+   *   postalCode: 'V1V1V1',
+   *   type: 'Home',
+   *   primary: true
+   * }
+   */
+  private $addresses;
```
Comment (trb) [inline src/LassoLead.php]: Not initialized as an array, might be an issue with line 151

File: signup.php [hunk 10]
```diff
@@ -3,35 +3,56 @@
...
-/* This is a security token and should be kept private
- * this should not be a hidden field on your registration page.
+/*
+ * This is a security token and should be kept private and should not be a
+ * hidden field on your registration page
  */
-$lassoUid  = '';
+$apiKey = '';
+
+if (empty($clientId) || empty($projectId) || empty($apiKey)) {
```
Comment (trb) [inline signup.php:18]: I _like_ this. Error messages should give hints how to fix said error.

**Pull request merged**